packages/tasit-action/src/TasitAct.js

```diff
@@ -0,0 +1,144 @@
+import "ethers/dist/shims.js";
+// Note: ethers SHOULD be imported from their main object
+// shims aren't injected with package import
+import { ethers } from "ethers";
+
+class Utils {
+  static isAddress = address => {
+    return typeof address === "string" && address.match(/^0x[0-9A-Fa-f]{40}$/);
+  };
+
+  static isABI = abi => {
+    return abi && Array.isArray(abi);
+  };
+
+  // https://github.com/ethers-io/ethers.js/blob/db383a3121bb8cf5c80c5488e853101d8c1df353/src.ts/utils/properties.ts#L20
+  static isEthersJsSigner = signer => {
+    return signer && signer._ethersType === "Signer";
+  };
+}
+
+class Subscription {
+  #txPromise;
+  #provider;
+
+  constructor(txPromise, provider) {
+    this.#txPromise = txPromise;
+    this.#provider = provider;
+  }
+
+  on = async (trigger, callback) => {
+    let blockHeight = 0;
+
+    if (trigger === "confirmation") blockHeight = 1;
+    else {
+      throw new Error(`Invalid subscription trigger, use: ["confirmation"]`);
+    }
+
+    if (!callback || typeof callback !== "function") {
+      throw new Error(`Cannot subscribe without a function`);
+    }
+
+    const tx = await this.#txPromise;
+
+    this.#provider.on(tx.hash, async receipt => {
```

Comment:
Does `ethers.js` expose any functions for blockchain reorgs due to uncles that remove older confirmations?

Reply:
Sorry, I didn't get it.

Reply:
This is related to my question about emitting an event when a certain number of confirmations isn't reached. https://github.com/tasitlabs/TasitSDK/pull/71#discussion_r245073386

An uncle is a block that seemed potentially valid but turned out not to be https://forum.ethereum.org/discussion/2262/eli5-whats-an-uncle-in-ethereum-mining

I'm not sure if I have the semantics 100% correct, but the thrust of my point is that sometimes a transaction might be confirmed once in a block but then never again when the blockchain has a tiny fork and the longest chain wins. Or 3 times but then never again and removed from the "true" chain.

Reply:
Extracted to https://github.com/tasitlabs/TasitSDK/issues/104